modules/internal/CPtr.chpl

```diff
@@ -223,8 +223,26 @@ module CPtr {
     :arg arr: the array for which we should retrieve a pointer
     :returns: a pointer to the array data
   */
-  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) {
-    if !arr._value.oneDData then halt("error: c_ptrTo(multi_ddata array");
+  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) && !arr._value.isDefaultRectangular() {
+    if !chpl__getActualArray(arr).oneDData then halt("error: c_ptrTo(multi_ddata array");
+    return c_pointer_return(arr[arr.domain.low]);
+  }
+
+  pragma "no doc"
+  inline proc c_ptrTo(arr: []) where arr._value.isDefaultRectangular() {
+    const val = arr._value;
+    if chpl__isArrayView(val) {
```

Comment:
If I'm reading the above right, array views on DRs return 'true' for 'isDefaultRectangular()' which seems surprising to me (but is also a pre-existing condition before this PR if I'm not mistaken).  I'd think that it should return false for array views and that the filter here should be more like "isDR || isArrayViewThatWrapsDR".

W.r.t. that note and your TODO in the code below, it seems to me that the better way to implement c_ptrTo() for arrays would be to have its body call a method on the array itself that would be implemented on _array in terms of a dsi routine that each domain map could them specialize.  That would put this logic where it belongs (in the array implementation) rather than way out here in CPtr.  It would also avoid resorting to a ._value access outside of the domain map code.